### ISSUE TITLE
eShopOnContainers no longer uses Autofac but TinyIoc ?

### ISSUE THREAD

**Issue by sushicodeur** [MicrosoftDocs/xamarin-docs#94]
Hello,

"There are many dependency injection containers available, with the eShopOnContainers mobile app using Autofac to manage the instantiation of view model and service classes in the app."

It looks like eShopOnContainers now uses TinyIoc : https://github.com/dotnet-architecture/eShopOnContainers/pull/465 

The code in this documentation can be misleading.

Well, this is for your informations, even if you don't plan to rewrite the whole chapter a mention that code is no longer the same could save time to your readers.

Thank you :)

Alain

---
#### Détails du document

⚠ *Ne pas modifier cette section. C’est obligatoire pour docs.microsoft.com ➟ Liaison des problèmes GitHub.*

* ID: 496aefc1-60e7-3484-12d1-ba383cfc099b
* Version Independent ID: 35133eba-8bc2-a54e-b341-d873f9a3c468
* Content: [Injection de dépendances - Xamarin](https://docs.microsoft.com/fr-fr/xamarin/xamarin-forms/enterprise-application-patterns/dependency-injection#feedback)
* Content Source: [docs/xamarin-forms/enterprise-application-patterns/dependency-injection.md](https://github.com/MicrosoftDocs/xamarin-docs.fr-FR/blob/live/docs/xamarin-forms/enterprise-application-patterns/dependency-injection.md)
* Service: **unspecified**
* GitHub Login: @davidbritch
* Microsoft Alias: **dabritch**

**Comment by davidbritch** [comment]
Hi Alain,

The eShop app does indeed now use TinyIoC instead of Autofac. However, this change has been made in the DEV branch for the app, rather than the MASTER branch. Once v2 of the app is complete, the DEV branch will be merged into MASTER, and an updated eBook/docs will be released.

**Comment by pstricks-fans** [comment]
I am wondering why you all did not use the generic `Microsoft.Extensions.DependencyInjection` here instead of TinyIOC. By the way, I also requested an example about it [here](https://github.com/xamarin/xamarin-forms-samples/issues/526).

**Comment by irperez** [comment]
I have to aggree with @pstricks-fans.  Is there a limitation with Microsoft.Extensions.DependencyInjection that we should be aware of?  

I was not planning on using Autofac.

And its also a bit weird that I see Autofac code inside a class library.  Isn't that a code smell?

Again, I'm not sure of the limitation, maybe its warranted.
